framework/src/modules/chain/block_processor_v2.js

```diff
@@ -281,19 +281,24 @@ class BlockProcessorV2 extends BaseBlockProcessor {
 		this.create.pipe([
 			// Create a block with with basic block and bft properties
 			async data => {
-				const previouslyForged = await this._getPreviouslyForgedMap();
+				const previouslyForgedMap = await this._getPreviouslyForgedMap();
 				const delegatePublicKey = data.keypair.publicKey.toString('hex');
+				const height = data.previousBlock.height + 1;
+				const previousBlockId = data.previousBlock.id;
 				const maxHeightPreviouslyForged =
-					previouslyForged[delegatePublicKey] || 0;
+					previouslyForgedMap[delegatePublicKey] || height;
```

Comment:
I am not sure if that's right. `data.previousBlock` refers to the previous block in the chain, not the the previous block forged by delegate. So in case we don't know what height delegated previously forged, setting it to previous block in the chain does not make sense.

Reply:
On the LIP it says
> The first block forged by the delegate uses the value max{A.height, A.maxHeightPreviouslyForged} in the maxHeightPreviouslyForged property.

Reply:
I think, this condition applies to migrate a node with existing data explicitly. 

The LIP also says very clearly that it should be `0` in case delegate didn't forged the block earlier. 

> If a delegate forges a block B to be added to the current chain, then B.maxHeightPreviouslyForged must be the largest height at which the delegate forging the block previously forged a block on any chain or 0 if the delegate did not forge a block before.

https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#additional-block-header-properties